tests/test_md.py

```diff
@@ -0,0 +1,497 @@
+        struct=single_point.struct,
+    )
+    assert dyn.ensemble == expected
+
+
+def test_npt():
+    """Test NPT molecular dynamics."""
+    restart_path_1 = Path("Cl4Na4-npt-T300.0-p1.0-res-2.xyz")
+    restart_path_2 = Path("Cl4Na4-npt-T300.0-p1.0-res-4.xyz")
+    traj_path = Path("Cl4Na4-npt-T300.0-p1.0-traj.xyz")
+    stats_path = Path("Cl4Na4-npt-T300.0-p1.0-stats.dat")
+
+    assert not restart_path_1.exists()
+    assert not restart_path_2.exists()
+    assert not traj_path.exists()
+    assert not stats_path.exists()
+
+    single_point = SinglePoint(
+        struct_path=DATA_PATH / "NaCl.cif",
+        architecture="mace",
+        calc_kwargs={"model": MODEL_PATH},
+    )
+    npt = NPT(
+        struct=single_point.struct,
+        pressure=1.0,
+        temp=300.0,
+        steps=4,
+        traj_every=1,
+        restart_every=2,
+        stats_every=1,
+    )
+
+    try:
+        npt.run()
+
+        restart_atoms_1 = read(restart_path_1)
+        assert isinstance(restart_atoms_1, Atoms)
+        restart_atoms_2 = read(restart_path_2)
+        assert isinstance(restart_atoms_2, Atoms)
+
+        traj = read(traj_path, index=":")
+        assert all(isinstance(image, Atoms) for image in traj)
+        assert len(traj) == 4
+
+        with open(stats_path, encoding="utf8") as stats_file:
+            lines = stats_file.readlines()
+            assert "Pressure[bar] | T [K]" in lines[0]
+            assert len(lines) == 5
+    finally:
+        restart_path_1.unlink(missing_ok=True)
+        restart_path_2.unlink(missing_ok=True)
+        traj_path.unlink(missing_ok=True)
+        stats_path.unlink(missing_ok=True)
+
+
+def test_nvt_nh():
+    """Test NVT-Nosé–Hoover  molecular dynamics."""
+    restart_path = Path("Cl4Na4-nvt-nh-T300.0-res-3.xyz")
+    traj_path = Path("Cl4Na4-nvt-nh-T300.0-traj.xyz")
+    stats_path = Path("Cl4Na4-nvt-nh-T300.0-stats.dat")
+
+    assert not restart_path.exists()
+    assert not traj_path.exists()
+    assert not stats_path.exists()
+
+    single_point = SinglePoint(
+        struct_path=DATA_PATH / "NaCl.cif",
+        architecture="mace",
+        calc_kwargs={"model": MODEL_PATH},
+    )
+    nvt_nh = NVT_NH(
+        struct=single_point.struct,
+        temp=300.0,
+        steps=3,
+        traj_every=1,
+        restart_every=3,
+        stats_every=1,
+    )
+
+    try:
+        nvt_nh.run()
+
+        restart_atoms = read(restart_path)
+        assert isinstance(restart_atoms, Atoms)
+
+        traj = read(traj_path, index=":")
+        assert all(isinstance(image, Atoms) for image in traj)
+        assert len(traj) == 3
+
+        with open(stats_path, encoding="utf8") as stats_file:
+            lines = stats_file.readlines()
+            assert " | T [K]" in lines[0]
+            assert len(lines) == 4
+    finally:
+        restart_path.unlink(missing_ok=True)
+        traj_path.unlink(missing_ok=True)
+        stats_path.unlink(missing_ok=True)
+
+
+def test_nve(tmp_path):
```

Comment:
I will add an extra test where we do not initialise the struicture from a single point but explicitly read the atoms and set the calculator.